kola/tests/misc/tpm.go

```diff
@@ -0,0 +1,222 @@
+package misc
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/flatcar/mantle/kola/cluster"
+	"github.com/flatcar/mantle/kola/register"
+	"github.com/flatcar/mantle/platform"
+	"github.com/flatcar/mantle/platform/conf"
+	"github.com/flatcar/mantle/platform/machine/qemu"
+	"github.com/flatcar/mantle/platform/machine/unprivqemu"
+	"github.com/flatcar/mantle/system/exec"
+	"github.com/flatcar/mantle/util"
+)
+
+const (
+	IgnitionConfigRootTPM = `{
+		"ignition": {
+			"config": {},
+			"timeouts": {},
+			"version": "3.3.0"
+		},
+		"kernelArguments": {
+			"shouldExist": ["rd.luks.name=12345678-9abc-def0-1234-56789abcdef0=rootencrypted", "systemd.mask=systemd-cryptsetup@rootencrypted.service"]
+		},
+		"networkd": {},
+		"storage": {
+			"disks": [
+				{
+					"device": "/dev/disk/by-id/virtio-secondary",
+					"wipeTable": true,
+					"partitions": [
+						{
+							"label": "root",
+							"number": 1
+						}
+					]
+				}
+			],
+			"luks": [
+				{
+					"name": "rootencrypted",
+					"device": "/dev/disk/by-partlabel/root",
+					"uuid": "12345678-9abc-def0-1234-56789abcdef0",
+					"clevis": {
+						"tpm2": true
+					}
+				}
+			],
+			"filesystems": [
+				{
+					"device": "/dev/disk/by-partlabel/ROOT",
+					"format": "none",
```

Comment:
What also works is to use this partition as luks partition (the `luks:` section need the entry to wipe the existing content), then we could drop the secondary disk. I think this is even more interesting to test.

Reply:
By the way, did you see the `needsNetwork` entry for clevis? I wonder if that triggers the network target in the initrd, didn't try it yet…

Reply:
> What also works is to use this partition as luks partition (the luks: section needs a wipeVolume entry to wipe the existing content), then we could drop the secondary disk. I think this is even more interesting to test.

Sounds good. I changed the cl.tang.root and cl.tpm.root to test encryption of the ROOT partition on the primary disk.

> By the way, did you see the custom: \n needsNetwork: entry for clevis? I wonder if that triggers the network target in the initrd, didn't try it yet…

Maybe you've already seen this, but once you set `custom.needsNetwork`, you also need to set `custom.pin` and `custom.config`. So, I'm not sure whether that would be a good alternative to how we currently do it in the tests, setting the `rd.networkd=1` cmdline parameter.

Reply:
Ok, didn't know that - and I also think that it might not help much because on second boot the initrd wouldn't know about this setting if it ends up in crypttab.